connect/transforms/src/main/java/org/apache/kafka/connect/transforms/field/SingleFieldPath.java

```diff
@@ -0,0 +1,500 @@
+                            field.append(pathText, start, idx - 1)
+                                .append(BACKTICK);
+
+                            idx++;
+                            start = idx;
+                        } else {
+                            // this backtick isn't followed by a dot; include it in the field name, but continue
+                            // looking for a matching backtick that is followed by a dot
+                            idx++;
+                        }
+                        continue;
+                    }
+
+                    boolean afterABackslash = pathText.charAt(idx - 1) == BACKSLASH;
+                    if (afterABackslash) {
+                        // this backtick was escaped; include it in the field name, but continue
+                        // looking for an unescaped matching backtick
+                        field.append(pathText, start, idx - 1)
+                            .append(BACKTICK);
+
+                        idx++;
+                        start = idx;
+                        continue;
+                    }
+                    // we've found our matching backtick
+                    field.append(pathText, start, idx);
+                    steps.add(field.toString());
+                    idx += 2; // increment by two to include the backtick and the dot after it
+                    break;
+                }
+
+
+            }
+        }
+        return steps.toArray(new String[0]);
+    }
+
+
+    /**
+     * Access a {@code Field} at the current path within a schema {@code Schema}
+     * If field is not found, then {@code null} is returned.
+     */
+    public Field fieldFrom(Schema schema) {
+        if (path.length == 1) {
+            return schema.field(path[0]);
+        } else {
+            Schema current = schema;
+            for (int i = 0; i < path.length; i++) {
+                if (current == null) {
+                    return null;
+                }
+                if (i == path.length - 1) { // get value
+                    return current.field(path[i]);
+                } else { // iterate
+                    current = current.field(path[i]).schema();
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Access a value at the current path within a schema-based {@code Struct}
+     * If object is not found, then {@code null} is returned.
+     */
+    public Object valueFrom(Struct struct) {
+        if (path.length == 1) {
+            return struct.get(path[0]);
+        } else {
+            Struct current = struct;
+            for (int i = 0; i < path.length; i++) {
+                if (current == null) {
+                    return null;
+                }
+                if (i == path.length - 1) { // get value
+                    return current.get(path[i]);
+                } else { // iterate
+                    current = current.getStruct(path[i]);
```

Comment:
Won't this cause a `DataException` if the field isn't found in this struct?

Reply:
Took a stab at this; it's a little unclean but all the tests pass; LMKWYT

```java
public Object valueFrom(Struct struct) {
    Struct current = struct;
    for (String pathSegment : Arrays.asList(path).subList(0, path.length - 1)) {
        // Check to see if the field actually exists
        if (current.schema().field(pathSegment) == null) {
            return null;
        }
        current = current.getStruct(pathSegment);
    }
    return current.get(path[path.length - 1]);
}
```

(It also feels like it might be a little more ergonomic if `path` were a `List` instead of an array, but that's not a blocker)

Reply:
Similar to the trie suggestion, I think a list may be simpler than an array. I gave it a try.